generate_tags.py

```diff
@@ -0,0 +1,100 @@
+import json
+
+import requests
+
+# This script loads the enterprise-attack.json mitre dataset into a dict (mitre_dataset)
+# according to the parameters set below. The dict is then iterated to produce an attack.py
+# file, containing variables for the multiple Mitre ATT&CK tags relevant to the project.
+#
+# The enterprise-attack.json file is sourced from Mitre's repository at
+# https://github.com/mitre-attack/attack-stix-data/tree/master/enterprise-attack.
+
+# Replace the URL below with the version of your choice.
+
+mitre_source = (
+    "https://raw.githubusercontent.com/mitre-attack/attack-stix-data/"
+    "master/enterprise-attack/enterprise-attack-14.1.json"
+)
+
+# Begin parameters; modify these to customize the tags produced by the script
+
+included_platforms = ["Windows", "Linux"]
+include_subtechniques = True
+include_deprecated = False  # include techniques marked as "deprecated"
+include_revoked = False  # include techniques marked as "revoked"
+
+# End parameters
+
+
+def retrieve_mitre_data(url):
+    print("Attempting download of Mitre data, please wait...")
+    try:
+        response = requests.get(url)
+        if response.status_code == 200:
+            print("\nDownload finished!")
+            return json.loads(response.content)
+        else:
+            print("Error: ", response.status_code)
+            return None
+    except requests.RequestException as e:
+        print("Error: ", e)
+        return None
```

Comment:
This function mixes how errors are returned. On the one hand, come errors are caught, such as `RequestException` and `None` is returned as a sentinel value to indicate an error has occurred. On the other, exceptions, such as `JSONDecodeError` are not handled here and are therefore raised to the caller.

1. Pick a single way to communicate errors to the caller. Either return `None` meaning, "I couldn't retrieve the data" or allow errors to be raised back to the caller.
2. Unless the function can take some action based on the error, it's best to raise it back to the caller. It's fine if you want to catch it and add some meaningful context in a print/log statement, but if the error won't cause the function to take some different action, there's not point in catching it here. Catch it somewhere where it's appropriate to make a decision about how to proceed. Or don't and let the program crash.